internal/repository/key.go

```diff
@@ -87,12 +87,12 @@ func OpenKey(ctx context.Context, s *Repository, name string, password string) (
 	}
 
 	// decrypt master keys
-	buf := make([]byte, len(k.Data))
-	n, err := k.user.Decrypt(buf, k.Data)
+	nonce, ciphertext := k.Data[:k.user.NonceSize()], k.Data[k.user.NonceSize():]
+	buf := make([]byte, 0, len(ciphertext))
+	buf, err = k.user.Open(buf, nonce, ciphertext, nil)
```

Comment:
Here and elsewhere you could skip the `make([]byte, 0, len(ciphertext))` and pass nil instead, as that will be done automatically.

Reply:
Good point, although I did only find this one occurrence. I've replaced it.